src/test/java/tsunami/security/scanner/utilities/FreeMarkerUtilTest.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package tsunami.security.scanner.utilities;
+
+import static com.google.common.truth.Truth.assertThat;
+import static com.google.common.truth.Truth8.assertThat;
+import static org.junit.Assert.assertThrows;
+
+import freemarker.template.TemplateException;
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class FreeMarkerUtilTest {
+
+  @Test
+  public void testFreeMarkerUtil() throws IOException, TemplateException {
+    Map<String, String> templateDataMap = new HashMap<>();
+    templateDataMap.put("jupyter_version", "0.0.3");
+
+    File configFile =
+        new File(System.getProperty("user.dir") + "/application/jupyter/jupyter.yaml");
+    String res = FreeMarkerUtil.replaceTemplates(templateDataMap, configFile);
```

Comment:
Avoid system properties within test if possible. For this test, you could use `TemporaryFolder` junit rule to create a temporary file for the template and use that instead.

And you don't need a real YAML file. A very short template file is good enough for testing.

Reply:
Document for `TemporaryFolder` rule: https://junit.org/junit4/javadoc/4.12/org/junit/rules/TemporaryFolder.html